numba/core/base.py

```diff
@@ -904,6 +904,15 @@ def call_internal_no_propagate(self, builder, fndesc, sig, args):
                                                    sig.args, args)
         return status, res
 
+    def call_overload(self, builder, impl, sig, args):
```

Comment:
I've written an almost identical `call_overload` locally as part of a large `ufunc` refactoring effort. I elected for it to take `sig.args` / an argument tuple opposed to a signature. When an overload is resolved, the return type is a function of the implementation, opposed to what is prescribed by a signature, this API perhaps suggests otherwise?